compiler/rustc_typeck/src/expr_use_visitor.rs

```diff
@@ -636,6 +709,12 @@ impl<'a, 'tcx> ExprUseVisitor<'a, 'tcx> {
                         place.projections.clone(),
                     );
 
+                    debug!("RFC2229 = expr_use_visitor.rs walk_captures maybe introduce fake_read");
+                    if body_owner_is_closure {
```

Comment:
this is not what we want. We want to created another loop that iterates `mc.typeck_results.fake_reads()`